src/views/design/ScriptsOverview/index.tsx

```diff
@@ -304,49 +309,58 @@ const ScriptsOverview = withStyles(styles)(
-                                            const scripts = getAsyncScripts(this.props.state);
-                                            const selectedScript = scripts.find((item) =>
-                                                getUniqueIdFromScript(item) === id);
-
-                                            redirectTo({
-                                                routeKey: ROUTE_KEYS.R_REPORTS,
-                                                queryParams: {
-                                                    script: selectedScript.name,
-                                                    version: selectedScript.version.number,
-                                                },
-                                            });
-                                        },
-                                    }, {
-                                        icon: <Delete />,
-                                        label: translator('scripts.overview.list.actions.delete'),
-                                        onClick: this.setScriptToDelete,
-                                    },
-                                ]}
+                                listActions={[].concat(
+                                    sessionStorage.getItem('authorities')
+                                        .includes('SCRIPT_EXECUTIONS_WRITE@PUBLIC') ? {
+                                            icon: <PlayArrowRounded />,
+                                            label: translator('scripts.overview.list.actions.execute'),
+                                            onClick: this.setScriptToExecute,
+                                        } : [],
+                                    sessionStorage.getItem('authorities')
+                                        .includes('SCRIPTS_WRITE@PUBLIC') ? {
+                                            icon: <Edit />,
+                                            label: translator('scripts.overview.list.actions.edit'),
+                                            onClick: (id: string) => {
+                                                const scripts = getAsyncScripts(this.props.state);
+                                                const selectedScript = scripts.find((item) =>
+                                                    getUniqueIdFromScript(item) === id);
+
+                                                redirectTo({
+                                                    routeKey: ROUTE_KEYS.R_SCRIPT_DETAIL,
+                                                    params: {
+                                                        name: selectedScript.name,
+                                                        version: selectedScript.version.number,
+                                                    },
+                                                });
+                                            },
+                                        } : [],
```

Comment:
if the authenticated user has the READ authority he/she should also be able to have a look at the script details. In this case, we can modify the pencil icon to an "eye" icon

Reply:
In addition to this, if a user with only read privileges is redirected to the script detail page, src/views/report/ScriptReportDetail/index.tsx should be adjusted to make all fields read-only